reactor-core/src/main/java/reactor/core/publisher/FluxTimeout.java

```diff
@@ -113,11 +136,29 @@ public void subscribe(CoreSubscriber<? super T> actual) {
 		TimeoutMainSubscriber(CoreSubscriber<? super T> actual,
 				Function<? super T, ? extends Publisher<V>> itemTimeout,
 				Publisher<? extends T> other,
-				String timeoutDescription) {
+				String timeoutDescription,
```

Comment:
same if other is not null then timeoutdescription should be null. Also we shouldn't build an exceptionMessage eagerly.